pkg/translator/prometheusremotewrite/metrics_to_prw_v2.go

```diff
@@ -0,0 +1,122 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package prometheusremotewrite // import "github.com/open-telemetry/opentelemetry-collector-contrib/pkg/translator/prometheusremotewrite"
+
+import (
+	"errors"
+	"fmt"
+	"strconv"
+
+	"github.com/prometheus/prometheus/model/labels"
+	writev2 "github.com/prometheus/prometheus/prompb/io/prometheus/write/v2"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+	"go.uber.org/multierr"
+
+	prometheustranslator "github.com/open-telemetry/opentelemetry-collector-contrib/pkg/translator/prometheus"
+)
+
+// FromMetricsV2 converts pmetric.Metrics to Prometheus remote write format 2.0.
+func FromMetricsV2(md pmetric.Metrics, settings Settings) (map[string]*writev2.TimeSeries, error) {
+	c := newPrometheusConverterV2()
+	errs := c.fromMetrics(md, settings)
+	tss := c.timeSeries()
+	out := make(map[string]*writev2.TimeSeries, len(tss))
+	for i := range tss {
+		out[strconv.Itoa(i)] = &tss[i]
+	}
+
+	return out, errs
+}
+
+// prometheusConverterV2 converts from OTLP to Prometheus write 2.0 format.
+type prometheusConverterV2 struct {
+	// TODO handle conflicts
+	unique      map[uint64]*writev2.TimeSeries
+	symbolTable writev2.SymbolsTable
+}
+
+func newPrometheusConverterV2() *prometheusConverterV2 {
+	return &prometheusConverterV2{
+		unique:      map[uint64]*writev2.TimeSeries{},
+		symbolTable: writev2.NewSymbolTable(),
+	}
+}
```

Comment:
Since prometheusconverter is a private object, users of this package need a way of retrieving the symbols table somehow.

Maybe FromMetricsV2 should also return that?

Reply:
Is the symbol table meant to be per-request, or long-lived?  But regardless it probably makes sense to return both, since both are needed in the request.

Reply:
It's per request. PRW is meant to be stateless

Reply:
then returning it makes sense to me.

Reply:
Will update the PR, should we also make sure it's empty before each request?

Reply:
FromMetrics() starts with newPrometheusConverterV2(), so I'm assuming it's impossible to not be empty, right?

Reply:
oh yeah your are right

Reply:
Addressed in new PR https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/35734